### PR TITLE
Add Contact API

### PR DESCRIPTION
**Pull request by h1z3y3**
Close https://github.com/micro/services/issues/311

### CODE REVIEW

File: contact/domain/contact.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	"fmt"
+	"time"
+
+	"github.com/micro/micro/v3/service/store"
+	"github.com/pkg/errors"
+
+	pb "github.com/micro/services/contact/proto"
+)
+
+type ContactIface interface {
```
Comment (asim) [inline contact/domain/contact.go:15]: I would either just call this Contact or ContactService

Comment (h1z3y3) [reply contact/domain/contact.go:15]: Actually it is in domain layer.
It used to inject to contact Service.

Or we just call it as `ContactDomain` ?

Comment (asim) [reply contact/domain/contact.go:15]: the package name is domain. So your usage is always `domain.Contact`, so `domain.ContactDomain` would be redundant. Also adding `Iface` to an interface isn't good practice, so I'd opt for brevity, its a `type of Contact interface` aka `type Contact interface`.

Comment (h1z3y3) [reply contact/domain/contact.go:15]: Make sense

File: contact/proto/contact.proto [hunk 2]
```diff
@@ -0,0 +1,109 @@
+syntax = "proto3";
+
+package contact;
+
+option go_package = "./proto;contact";
+
+service Contact {
+  rpc Create(CreateRequest) returns (CreateResponse) {}
```
[the remark this answers is not in the document]
Comment (h1z3y3) [reply contact/proto/contact.proto]: My IDEA help me format, I've modify the configuration.

**Comment by h1z3y3** [comment]
@asim  It's ready to merge, please do more review.

What's more than last review:

1. add some necessary comments;
2. add example.json and publicapi.json
3. resolve the code review conversations

What's missing:
* `Search()`: I'm gonna waiting for the SearchAPI as you said earlier.

File: contact/publicapi.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "contact",
+  "icon": "👥",
+  "category": "communication",
+  "display_name": "Contact"
```
Comment (asim) [inline contact/publicapi.json:5]: Contacts

File: contact/proto/contact.proto [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	// the birthday
+	string birthday = 6;
+	// the address
+	repeated Address addresses = 7;
+	// the social media username
+	repeated SocialMedia social_medias = 8;
+	// note of the contact
+	string note = 9;
+	// create date in timestamp
+	int64 created_at = 10;
```
Comment (asim) [inline contact/proto/contact.proto:72]: s/created_at/created
s/updated_at/updated

Can you make these string and then use `time.Now().Format(time.RFC3339)`

**Pull request merged**